### DISCUSSION TITLE
Next to the global isFetching and isMutating booleans, I would like a global lastError object

### DISCUSSION THREAD

**Discussion by Evertt** [DamianOsipiuk/vue-query#149] [Ideas]
I don't want to handle errors on a per-query basis. I want to handle errors globally. Simply by showing a toast / notification to the user saying something along the lines of "Sorry, something went wrong...". Would you be willing to add something like that. Ideally the lastError is not a boolean, but an object with details about the error. Namely, was it a fetching error, or a mutating error? With which key was the error associated, if any? Things like that.

That way I can write a single global error handler that displays the most useful message to the user possible.

**Comment by DamianOsipiuk** [comment]
Hi,

This is already possible when setting up the `VueQueryPlugin`.

This should work for you:

```ts
const queryCache = new QueryCache({
  onError: (error, query) => {
    console.log(error); // handle global error
  },
});

const vueQueryPluginOptions: VueQueryPluginOptions = {
  queryClientConfig: { queryCache },
};

const app = createApp(App)
  .use(VueQueryPlugin, vueQueryPluginOptions)
  .mount("#app");
```

**Reply by Evertt** [reply]
Is this just for queries? Or also for mutations? 😇

**Reply by DamianOsipiuk** [reply]
You can do the same thing with `MutationCache`
